homeassistant/core.py

```diff
@@ -180,6 +180,28 @@ class HassJobType(enum.Enum):
     Executor = 3
 
 
+class Cancel:
+    """Helper to manage cancelling subscriptions etc."""
+
+    def __init__(self, cancel: CALLBACK_TYPE) -> None:
+        """Initialize the cancel object."""
+        self._cancel = cancel
+        self.cancelled = False
+
+    @callback
+    def __call__(self) -> None:
+        """Backwards compatibility."""
+        self._cancel()
+
+    @callback
+    def cancel(self) -> None:
+        """Cancel."""
+        if self.cancelled:
+            return
+        self.cancelled = True
+        self._cancel()
+
+
```

Comment:
You should set _camcel to none here, otherwise the callback can't be garbage collected if something still holds on to the object.